status-im/status-react
# Fix replies & pinned messages

Both replies and pinned messages relied on subscribing their data from `messages`.


This worked only as long as we loaded the message in the database, so it would break say if another user replied to a message that wasn't in the current user view.

It also changes the way pinned messages are handled, before the notification was actually sent over the wire, but that's unnecessary, since it can be generated locally on both parts.
This is a bit of a breaking change with the previous version, since if you pin a message with this version, older version will not see a system message. this can be easily fixed by restoring the previous behavior of sending the message, but not sure it's worth it.

it also adds the ability to Delete message for everyone that have Deleted for me (discussed with John) and the ability of unpin messages that have been deleted for me.

I will add a couple of unit tests to validate the new features, but in the meantime I would like to get it through the testing pipeline as likely there will be some issues to address.

status-go PR https://github.com/status-im/status-go/pull/3414

I wanted to move the UI code, but it's already quite a few changes so I wanted to minimize disruptions.

Area to tests (communities/group-chat/1-to-1):

1) Pin messages
2) Pin old messages
3) Reply to messages
4) Reply to old messages
5) Delete replied to messages, both locally and for everyone
6) Delete pinned messages, both locally and everyone


I'll try to address this in this PR, but I think it should be ok to address separately, if qa is happy with it.
